### ISSUE TITLE
Loader: Handle missing components with better failure

### ISSUE THREAD

**Issue by humancompanion** [EightShapes/Blocks#89]
### Problem

Currently when the loader fails to load a component, say when someone unceremoniously removes the titlearea component and every page in your project breaks, the loader fails to every reach the stage by which it would inject component JS:

````
// Once all of the kids are done we'll spawn all JS
      if (self.child_count === self.children_loaded) {
        self.injectComponentJS();
      }
````

This needs to be addressed. My inclination here is to fail fast and not continue at all. Failing to load a component is a big red flag that probably results in an alert and an immediate bail.

### Resolution

We met back in December and agreed to do the following:

* [x] Fix logging to be level/threshold based
* [x] Add a subtle badge to upper right corner to indicate issues above a particular level

In other words, the default logging level should not be debug (currently it is just on/off) however, it should warn the user of some problems loading a page and failing to load a component is one of those issues.

**Comment by humancompanion** [comment]
@kevinmpowell's pull request fixed this up.
